src/nodeServer/models.js

```diff
@@ -1,27 +1,36 @@
 const fs = require('fs');
+const etag = require('etag');
+const path = require('path');
 const redisClient = require('./redisClient');
 const dbHelpers = require('../../db/dbHelpers');
 
 module.exports = {
   serveHtml: (request, response) => {
-    const htmlStream = fs.createReadStream('../client/index.html');
+    const htmlStream = fs.createReadStream(path.join(__dirname, '../client/index.html'));
     response.writeHead(200, { 'Content-Type': 'text/html' });
     htmlStream.pipe(response);
   },
-  serveBundle: (request, response) => {
-    const bundleStream = fs.createReadStream('../public/bundle.js');
-    response.writeHead(200, { 'Content-Type': 'text/javascript' });
-    bundleStream.pipe(response);
+  serveBundle: (request, response, fileName) => {
+    const bundleStream = fs.createReadStream(path.join(__dirname, `../public/${fileName}`));
+    let data = '';
+    bundleStream.on('data', (chunk) => {
+      data += chunk;
+    }).on('end', () => {
+      response.setHeader('ETag', etag(data));
+      response.writeHead(200, { 'Content-Type': 'text/javascript' });
+      response.end(data);
+    });
   },
   menuType: (request, response, itemInfo) => {
+    // keeping just incase of ACAO issues
```

Comment:
What is an ACAO issue?

Reply:
It "Access-Control-Allow-Origin", but now that I think of it Cross Origin Issue would be more clear. Bad naming on my part, I'll change it to Cross Origin instead God willing.